source/server/active_tcp_socket.cc

```diff
@@ -88,15 +114,40 @@ void ActiveTcpSocket::continueFilterChain(bool success) {
           // break the loop but should not create new connection
           no_error = false;
           break;
+        } else if ((*iter_)->maxReadBytes() == 0) {
+          // The filter return stop the filters iteration and doesn't need to inspect data.
+          // We consider it as the filter reject the connection.
+          socket_->ioHandle().close();
+          continueFilterChain(false);
```

Comment:
note, if the filter's onAccept() return StopIteration, then envoy will stop the iteration. Now I changed this behavior, if the filter return StopIteration and doesn't expect any data, it means the filter reject the connection.

Reply:
In the case that the filter returns StopIteration, pleas add `ASSERT(filter->maxReadBytes() > 0)`. And also the return value from `onData()`, if it is StopIteration, add `ASSERT(filter->maxReadBytes() > buffer->len())` or similar, to try to catch logic errors in the filters.

Reply:
got it, thanks!

Reply:
on second thought I should be able to allow the listener filter to return `StopIteration` from `onAccept`, it allows the filter to reject the connection based on its own rule.

Reply:
> In the case that the filter returns StopIteration, pleas add `ASSERT(filter->maxReadBytes() > 0)`. 

On second thought, we should be able to allow the filter to reject the connection by the `onAccept` based its own rule. There may have limited knowledge about the connection, but for example, it can reject the connection based on source address for some kind of access control.

> And also the return value from `onData()`, if it is StopIteration, add `ASSERT(filter->maxReadBytes() > buffer->len())` or similar, to try to catch logic errors in the filters.

I can't add this assert, since there may have multiple filters need the data, but expect different size. The buffer size is the max size in all the filters. So there may have the case the current filter's maxReadBytes() less than buffer's size.

Reply:
got it, let me update it.